docs/root/intro/arch_overview/security/threat_model.rst

```diff
@@ -15,16 +15,31 @@ highest priority concerns. Availability, in particular in areas relating to DoS
 exhaustion, is also a serious security concern for Envoy operators, in particular those utilizing
 Envoy in edge deployments.
 
-The Envoy availability stance around CPU and memory DoS, as well as Query-of-Death (QoD), is still
-evolving. We will continue to iterate and fix well known resource issues in the open, e.g. overload
-manager and watermark improvements. We will activate the security process for disclosures that
-appear to present a risk profile that is significantly greater than the current Envoy availability
-hardening status quo. Examples of disclosures that would elicit this response:
-
-* QoD; where a single query from a client can bring down an Envoy server.
-
-* Highly asymmetric resource exhaustion attacks, where very little traffic can cause resource exhaustion,
-  e.g. that delivered by a single client.
+We will activate the security release process for disclosures that meet the following criteria:
+
+* All issues that lead to loss of data confidentiality or integrity trigger the security release process.
+* An availability issue needs to meet all of the following criteria to trigger the security release process:
+  
+  - A component tagged as hardened is affected (see `Core and extensions`_ for the list of hardened components).
+    
+  - The type of traffic (upstream or downstream) that exhibits the issue matches component's hardening tag.
+    I.e. component tagged as “hardened to untrusted downstream” is affected by downstream request.
+    
+  - A resource exhaustion issue needs to meet these additional criteria:
+    
+    + Not covered by an existing timeout or where applying short timeout values is impractical and either
+      
+    + Memory exhaustion with amplification factor of 100 or more. I.e. 100 byte client request leading to
+      10000 bytes of memory consumed by Envoy or
+      
+    + CPU exhaustion where 100 or fewer HTTP/1 connections or HTTP/2 streams consume an entire CPU core.
```

Comment:
A single very high throughput connection can saturate a full CPU core.  Worth saying something about bitrate?

Reply:
Added 50Mbit/s requirement.

Reply:
What's the expected throughput of an Envoy?  We probably want attack traffic to result in at least a 2x or 4x slowdown.

Reply:
It depends on a lot of things, e.g. if TLS is enabled or not, the request size and so on. I think as phrased right now, if we had 100 connections it might be possible to easily saturate a core with 1 byte requests.

Unless what is being said is that we have a steady state 100 streams and are pumping traffic through at this rate? This seems really specific though, what about CPU exhaustion via other means?

I think the most interesting way to measure CPU exhaustion is the ratio of the work the client must do vs. the server. If this is highly asymmetric, then you have a problem.

Reply:
Makes sense. As discussed CPU exhaustion is expressed in client vs Envoy terms.